ios-apps-master/Beer_Diary/Beer_Diary/Add_Beer.swift

```diff
@@ -0,0 +1,256 @@
+    var date = NSDate()
+}
+
+func saveData()
+{
+    var tname = [String]()
+    var tbrewery = [String]()
+    var tstyle = [String]()
+    var tcountry = [String]()
+    var tthoughts = [String]()
+    var taroma = [Int]()
+    var tappearance = [Int]()
+    var tpalate = [Int]()
+    var ttaste = [Int]()
+    var tscore = [Float]()
+    var tdate = [NSDate]()
+    
+    
+    for x in BeerList
+    {
+        tname.append(x.name)
+        tbrewery.append(x.brewery)
+        tstyle.append(x.style)
+        tcountry.append(x.country)
+        tthoughts.append(x.thoughts)
+        taroma.append(x.aroma)
+        tappearance.append(x.appearance)
+        tpalate.append(x.palate)
+        ttaste.append(x.taste)
+        tscore.append(x.overall)
+        tdate.append(x.date)
+    }
+    
+    UserDefaults.standard.set(tname, forKey: "name")
+    UserDefaults.standard.set(tbrewery, forKey: "brewery")
+    UserDefaults.standard.set(tstyle, forKey: "style")
+    UserDefaults.standard.set(tcountry, forKey: "country")
+    UserDefaults.standard.set(tthoughts, forKey: "thoughts")
+    UserDefaults.standard.set(taroma, forKey: "aroma")
+    UserDefaults.standard.set(tappearance, forKey: "appearance")
+    UserDefaults.standard.set(tpalate, forKey: "palate")
+    UserDefaults.standard.set(ttaste, forKey: "taste")
+    UserDefaults.standard.set(tscore, forKey: "score")
+    UserDefaults.standard.set(tdate, forKey: "date")
+}
+
+class Add_Beer: UIViewController {
+ 
+    @IBOutlet weak var beerName: UITextField!
+    @IBOutlet weak var breweryName: UITextField!
+    @IBOutlet weak var styleName: UITextField!
+    @IBOutlet weak var countryName: UITextField!
+    @IBOutlet weak var aroma: UILabel!
+    @IBOutlet weak var appearance: UILabel!
+    @IBOutlet weak var palate: UILabel!
+    @IBOutlet weak var taste: UILabel!
+    @IBOutlet weak var overall: UILabel!
+    @IBOutlet weak var thoughts: UITextView!
+    @IBOutlet weak var aromaSlider: UISlider!
+    @IBOutlet weak var appearanceSlider: UISlider!
+    @IBOutlet weak var palateSlider: UISlider!
+    @IBOutlet weak var tasteSlider: UISlider!
+    
+    
+    var newItem = BeerItem()
+    
+    
+    
+    
+    @IBAction func sliderAroma(_ sender: Any)
+    {
+        let tempVal = Int(aromaSlider.value)
+        aromaSlider.value = Float(tempVal)
+        newItem.aroma = tempVal
+        aroma.text = "Aroma - \(tempVal)/10"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    @IBAction func sliderAppearance(_ sender: Any)
+    {
+        let tempVal = Int(appearanceSlider.value)
+        appearanceSlider.value = Float(tempVal)
+        newItem.appearance = tempVal
+        appearance.text = "Appearance - \(tempVal)/5"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
```

Comment:
I would extract this line to a method, because it's complex and copy pasted all over. E.g. `updateOverallScore()`